mne/preprocessing/xdawn.py

```diff
@@ -0,0 +1,565 @@
+            if reg == 'lws':
+                try:
+                    from sklearn.covariance import LedoitWolf
+                except ImportError:
+                    raise Exception('the scikit-learn package is missing '
+                                    'and required for regularization.')
+                # init sklearn.covariance.LedoitWolf estimator
+                skl_cov = LedoitWolf(store_precision=False,
+                                     assume_centered=True)
+            elif reg == 'oas':
+                try:
+                    from sklearn.covariance import OAS
+                except ImportError:
+                    raise Exception('the scikit-learn package is missing '
+                                    'and required for regularization.')
+                # init sklearn.covariance.OAS estimator
+                skl_cov = OAS(store_precision=False,
+                              assume_centered=True)
+            else:
+                raise ValueError("regularization parameter should be "
+                                 "of type str (got %s)." % type(reg))
+        else:
+            raise ValueError("regularization parameter should be "
+                             "of type str (got %s)." % type(reg))
+
+        # compute regularized covariance using sklearn
+        covar = skl_cov.fit(data.T).covariance_
+
+    return covar
+
+
+def _check_overlapp(epochs):
+    """check if events are overlapped."""
+    isi = np.diff(epochs.events[:, 0])
+    window = int((epochs.tmax - epochs.tmin) * epochs.info['sfreq'])
+    return isi.min() < window
+
+
+def _construct_signal_from_epochs(epochs):
+    """Reconstruct pseudo continuous signal from epochs."""
+    start_ix = (np.min(epochs.events[:, 0])
+                + int(epochs.tmin * epochs.info['sfreq']))
+    end_ix = (np.max(epochs.events[:, 0])
+              + int(epochs.tmax * epochs.info['sfreq']) + 1)
+
+    ns = end_ix - start_ix
+    ns_epochs = epochs._data.shape[2]
+    ne = epochs._data.shape[1]
+    ix_events = epochs.events[:, 0] - epochs.events[0, 0]
+
+    data = np.zeros((ne, ns))
+    for i, ix in enumerate(ix_events):
+        data[:, ix:(ix + ns_epochs)] = epochs._data[i]
+
+    return data
+
+
+def least_square_evoked(epochs, return_toeplitz=False):
+    """Least square estimation of evoked response from a Epochs instance.
+
+    Parameters
+    ----------
+    epochs : Epochs instance
+        An instance of Epochs.
+    return_toeplitz : bool
+        If true, compute the toeplitz matrix.
+
+    Returns
+    -------
+    evokeds : dict of evoked instance
+        An dict of evoked instance for each event type in epochs.event_id.
+    toeplitz : dict of array
+        If return_toeplitz is true, return the toeplitz matrix for each event
+        type in epochs.event_id.
+    """
+    if not isinstance(epochs, _BaseEpochs):
+        raise ValueError('epochs must be an instance of `mne.Epochs`')
+
+    evs = epochs.events.copy()
```

Comment:
even though your code is pretty clean and I don't have issues with your variable name if they were isolated code. But I feel if it's part of a repo, it should be as consistent with the repo norms as possible. That makes life easy for a new developer ...